animations-dbus/animations-dbus-server-object.c

```diff
@@ -471,6 +473,7 @@ on_got_session_bus_connection (GObject      *object G_GNUC_UNUSED,
   if (connection == NULL)
     {
       g_task_return_error (task, g_steal_pointer (&local_error));
+      g_object_unref (task);
```

Comment:
This (and the one below) look like valid fixes. I have a slightly more comprehensive patch which cleans up the code a bit (I wrote it while trying to understand the reference count transfers), which you might want to use/incorporate/ignore depending on whether you think it’s correct. I haven’t tested it (and since I’m just doing a drive-by review here, I have no plans to take it any further):

https://github.com/endlessm/libanimations-dbus/tree/T29817-refcounting

Reply:
(Outdated.)